meta-psa/recipes-security/psa-storage/psa-storage.bb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+SUMMARY = "ARM Platform Security Architecture (PSA) Protected Storage Library"
+DEPENDS = ""
+HOMEPAGE = "https://github.com/ARMmbed/psa-storage"
+DESCRIPTION = "A Linux C library reference implementation of the PSA protected_storage.h API"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = " \
+    file://${WORKDIR}/git/LICENSE;md5=302d50a6369f5f22efdb674db908167a \
+    file://${WORKDIR}/git/apache-2.0.txt;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+    "
+
+SRC_URI = "git://git@github.com/armmbed/psa-storage.git;protocol=ssh;nobranch=1"
+SRCREV = "3572fc55c8cc494bfeb442220f8a91542ed7bb99"
+
+PV = "1.0.0+git${SRCPV}"
+S = "${WORKDIR}/git"
+
+# Standard make options
+EXTRA_OEMAKE = "'CC=${CC}' 'RANLIB=${RANLIB}' 'AR=${AR}' 'CFLAGS=${CFLAGS}' 'BUILDDIR=${S}'"
+
+FILES_${PN} = "${bindir}/ ${includedir}/ ${libdir}/"
+# Work around libpsastorage.so.3 symlink error by not including files in dev package.
+FILES_${PN}-dev = ""
```

Comment:
Where did the number `3` in `libpsastorage.so.3` come from? Is it a relic of mbed-crypto?

I suggest that we actually want psa-storage to:
* Create a file `libpsastorage.so.${major}.${minor}`
* In the `install` `lib/makefile` target, add a symlink from `libpsastorage.so.${major}` to `libpsastorage.so.${major}.${minor}` and add a symlink from `libpsastorage.so` to `libpsastorage.so.${major}`.
* Put the `libpsastorage.so` symlink in the `-dev` package.
* Put `libpsastorage.so.${major}.${minor}` and the `libpsastorage.so.${major}` symlink in the default package.

I *think* the default values of `FILES_${PN}` and `FILES_${PN}-dev` will just put the library and its symlinks in the correct packages if they're installed in `${libdir}`

The example app in psa-storage should link against `libpsastorage.so.${major}`.

Sources:
* http://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html
* https://www.yoctoproject.org/docs/latest/mega-manual/mega-manual.html#var-FILES_SOLIBSDEV
* https://www.yoctoproject.org/docs/latest/mega-manual/mega-manual.html#var-SOLIBSDEV
* https://lists.yoctoproject.org/pipermail/yocto/2011-November/003542.html

Reply:
I'm in accordance with @jonathanhaigh-arm.

Reply:
Fixed in https://github.com/ARMmbed/meta-mbl/pull/619

Reply:
I suppose the recipe has been fixed to work with the current source code. The source code still doesn't version the library though, so https://jira.arm.com/browse/IOTMBL-2414.

Reply:
Ignore my last comment - Sorry @simonqhughes, I didn't realize you'd already fixed this in the psa storage build system.